src/fu-self-test.c

```diff
@@ -645,6 +645,7 @@ fu_engine_requirements_func(gconstpointer user_data)
 			   "  <requires>"
 			   "    <id compare=\"ge\" version=\"1.2.3\">org.test.dummy</id>"
 			   "    <hardware>6ff95c9c-ae41-5f59-9d90-3ec1ea66091e</hardware>"
+			   "    <not_hardware>ffffffff-ffff-ffff-ffff-ffffffffffff</not_hardware>"
```

Comment:
I think you need a test for what happens when hardware and not hardware have the same HWID.  It shouldn't work, but it's a border case that should always be checked in case of future logic changes.

Reply:
Good point, but I think it's more generic than that that even -- I don't think it makes sense to use `not_hardware` at all if specifying `hardware` -- and vice-versa. I've added a commit to that effect here.